Packages/CommonUI/CommonUI/Classes/View/MessageView/ZoomableScrollView.swift

```diff
@@ -0,0 +1,155 @@
+	
+	init(@ViewBuilder content: () -> Content) {
+		self.content = content()
+	}
+	
+	@State var doubleTap = PassthroughSubject<Void, Never>()
+	
+	var body: some View {
+		ZoomableScrollViewImpl(
+			content: content,
+			doubleTap: doubleTap.eraseToAnyPublisher()
+		)
+		.onTapGesture(count: 2) {
+			doubleTap.send()
+		}
+	}
+}
+
+private struct ZoomableScrollViewImpl<Content: View>: UIViewControllerRepresentable {
+	let content: Content
+	let doubleTap: AnyPublisher<Void, Never>
+	
+	func makeUIViewController(context: Context) -> ZoomableScrollViewController {
+		ZoomableScrollViewController(coordinator: context.coordinator, doubleTap: doubleTap)
+	}
+	
+	func makeCoordinator() -> Coordinator {
+		Coordinator(hostingController: UIHostingController(rootView: content))
+	}
+	
+	func updateUIViewController(_ viewController: ZoomableScrollViewController, context: Context) {
+		viewController.update(content: content, doubleTap: doubleTap)
+	}
+	
+	// MARK: - ZoomableScrollViewController
+
+	class ZoomableScrollViewController: UIViewController, UIScrollViewDelegate {
+		
+		let coordinator: Coordinator
+		let scrollView = UIScrollView()
+		
+		var doubleTapCancellable: Cancellable?
+		var updateConstraintsCancellable: Cancellable?
+		
+		private var hostedView: UIView { coordinator.hostingController.view! }
+		
+		private var contentSizeConstraints: [NSLayoutConstraint] = [] {
+			willSet { NSLayoutConstraint.deactivate(contentSizeConstraints) }
+			didSet { NSLayoutConstraint.activate(contentSizeConstraints) }
+		}
+		
+		@available(*, unavailable)
+		required init?(coder: NSCoder) { fatalError() }
+		init(coordinator: Coordinator, doubleTap: AnyPublisher<Void, Never>) {
+			self.coordinator = coordinator
+			super.init(nibName: nil, bundle: nil)
+			view = scrollView
+			
+			scrollView.delegate = self
+			scrollView.maximumZoomScale = 10
+			scrollView.minimumZoomScale = 1
+			scrollView.bouncesZoom = true
+			scrollView.showsHorizontalScrollIndicator = false
+			scrollView.showsVerticalScrollIndicator = false
+			scrollView.clipsToBounds = false
+			
+			let hostedView = coordinator.hostingController.view!
+			hostedView.translatesAutoresizingMaskIntoConstraints = false
+			hostedView.backgroundColor = UIColor.black
+			scrollView.addSubview(hostedView)
+			NSLayoutConstraint.activate([
+				hostedView.leadingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.leadingAnchor),
+				hostedView.trailingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.trailingAnchor),
+				hostedView.topAnchor.constraint(equalTo: scrollView.contentLayoutGuide.topAnchor),
+				hostedView.bottomAnchor.constraint(equalTo: scrollView.contentLayoutGuide.bottomAnchor)
+			])
+			
+			updateConstraintsCancellable = scrollView.publisher(for: \.bounds).map(\.size).removeDuplicates()
+				.sink { [unowned self] _ in
+					view.setNeedsUpdateConstraints()
+				}
+			doubleTapCancellable = doubleTap.sink { [unowned self] in handleDoubleTap() }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/CommonUI/CommonUI/Classes/View/MessageView/ZoomableScrollView.swift#L95"></span>It's safer to use weak instead of unowned</td>
    </tr>
  </tbody>
</table>